image_view/src/nodes/video_recorder.cpp

```diff
@@ -66,33 +73,68 @@ void callback(const sensor_msgs::ImageConstPtr& image_msg)
             exit(-1);
         }
 
-        ROS_INFO_STREAM("Starting to record " << codec << " video at " << size << "@" << fps << "fps. Press Ctrl+C to stop recording." );
+        ROS_INFO_STREAM("Starting to record " << codec << " video at " << size << "@" << fps << "fps. Press Ctrl+C to stop recording."<< "\x1b[1F" );
 
     }
-
-    if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration(1 / fps))
-    {
-      // Skip to get video with correct fps
-      return;
-    }
-
+        
     try
     {
       cv_bridge::CvtColorForDisplayOptions options;
       options.do_dynamic_scaling = use_dynamic_range;
       options.min_image_value = min_depth_range;
       options.max_image_value = max_depth_range;
       options.colormap = colormap;
-      const cv::Mat image = cv_bridge::cvtColorForDisplay(cv_bridge::toCvShare(image_msg), encoding, options)->image;
-      if (!image.empty()) {
-        outputVideo << image;
-        ROS_INFO_STREAM("Recording frame " << g_count << "\x1b[1F");
-        g_count++;
-        g_last_wrote_time = image_msg->header.stamp;
-      } else {
-          ROS_WARN("Frame skipped, no data!");
+      const cv::Mat imageCapture = cv_bridge::cvtColorForDisplay(cv_bridge::toCvShare(image_msg), encoding, options)->image;
+        
+      if(!imageCapture.empty())
+      {
+        if(g_count==0)
+        {
+          outputVideo << imageCapture;
+
+          g_count++;
+           
+          ROS_INFO_STREAM("Recording frame " << g_count << "\x1b[A");  
+          g_start_time = ros::Time::now(); 
+
+        }
+        else{
+          float elapsedTime = (ros::Time::now() - g_start_time).toSec();
+          int missingFrame = (elapsedTime*fps) - g_count;
+
+          if(missingFrame<=0){
+            imageBuff = cv_bridge::cvtColorForDisplay(cv_bridge::toCvShare(image_msg), encoding, options)->image.clone();
+              return; //skip frame
+          }
+              
+
+          int imagePad = missingFrame-1;
+
+          for( int x = 0; x<imagePad;x++)
```

Comment:
use a formatter to give something proper with spaces like: for(int x=0; x<imagePad; ++x)